apps/viewer/src/app.tsx

```diff
@@ -72,17 +51,48 @@ function App() {
   }, []);
 
   useEffect(() => {
-    const newTrackSourcesStates = new Map(trackSourcesStates);
-    remoteTrackSources.forEach((source) => {
-      if (!newTrackSourcesStates.get(source.sourceId)) newTrackSourcesStates.delete(source.sourceId);
-    });
-    if (newTrackSourcesStates.size !== trackSourcesStates.size) {
-      setTrackSourcesStates(newTrackSourcesStates);
+    if (remoteTrackSources.size && !hasInitialisedMainStreamRef.current) {
+      const [[firstSourceId]] = remoteTrackSources;
+
+      setMainSourceId(firstSourceId);
+
+      projectToMainStream(firstSourceId);
+
+      hasInitialisedMainStreamRef.current = true;
+    }
+  }, [remoteTrackSources.size]);
+
+  const handleClickVideo = (sourceId: SourceId) => {
+    projectToMainStream(sourceId, mainSourceId);
+
+    setMainSourceId(sourceId);
+  };
+
+  const mainSource = mainSourceId !== undefined ? remoteTrackSources.get(mainSourceId) : undefined;
+
+  const mainSourceSettings = () => {
```

Comment:
So the settings of main source are depending on `mainSourceId`,  we could use `useMemo` to cache it?